src/augur-node/dispatch-json-rpc-response.js

```diff
@@ -6,17 +6,28 @@ var isObject = require("../utils/is-object");
 
 function dispatchJsonRpcResponse(err, jsonRpcResponse) {
   if (err) throw err;
-  if (!jsonRpcResponse || !isObject(jsonRpcResponse) || jsonRpcResponse.id === undefined) {
+  if (!jsonRpcResponse || !isObject(jsonRpcResponse) || jsonRpcResponse.id === undefined || (jsonRpcResponse.id === null && (jsonRpcResponse.result === null && jsonRpcResponse.result.subscription === undefined))) {
     throw new Error("Bad JSON RPC response received:" + JSON.stringify(jsonRpcResponse));
   }
-  var callback = augurNodeState.getCallback(jsonRpcResponse.id);
+  var callback, result;
+  if (jsonRpcResponse.id !== null) {
+    callback = augurNodeState.getCallback(jsonRpcResponse.id);
+    augurNodeState.removeCallback(jsonRpcResponse.id);
+    result = jsonRpcResponse.result;
+  } else if (jsonRpcResponse.result.subscription !== null) {
+    callback = augurNodeState.getCallback("event:" + jsonRpcResponse.result.subscription);
+    result = jsonRpcResponse.result.result;
+    console.log(callback, result);
+  }
```

Comment:
Should there be an `else` here with "Bad JSON response received"?

Reply:
This should be handled above with that big IF block, but I think its cleaner to do in here. Will change